### PR TITLE
lnwallet+contractcourt: gracefully handle auto force close post data …

### PR DESCRIPTION
**Pull request by Roasbeef**
…loss

In this commit, update the start up logic to gracefully handle a seemingly rare case. In this case, a peer detects local data loss with a set of active HTLCs. These HTLCs then eventually expire (they may or may not actually "exist"), causing a force close decision. Before this PR, this attempt would fail with a fatal error that can impede start up.

To better handle such a scenario, we'll now catch the error when we fail to force close due to entering the DLP and instead terminate the state machine at the broadcast state. When a commitment transaction eventually confirms, we'll play it as normal.

Fixes https://github.com/lightningnetwork/lnd/issues/7984

### CODE REVIEW

File: contractcourt/channel_arbitrator.go
Comment (yyforyongyu) [inline contractcourt/channel_arbitrator.go:1064]: not sure about this step. Since we failed to create and broadcast the FC tx here, it means we should abort the `stateStep` as the arbitrator will now have undefined behaviors when it tries to create resolutions for under this state?

File: lnwallet/channel.go
Comment (yyforyongyu) [inline lnwallet/channel.go:7325]: 👍

**Review by yyforyongyu — COMMENTED** [review]
I think we should catch the error here to make sure we can finish startup,
https://github.com/lightningnetwork/lnd/blob/90effda090e752662500aa47eaed8b8619687e4e/contractcourt/channel_arbitrator.go#L534

And no advancing the state in channel arbitrator. Once data loss is detected, we'll wait for the remote to force close it. Then the chain arbitrator will detect it and send the event to `RemoteUnilateralClosure`, which will be picked up and handled by the channel arbitrator to clean up the resolutions.

File: contractcourt/channel_arbitrator.go
Comment (Crypt-iQ) [reply contractcourt/channel_arbitrator.go:1064]: I agree, I think we should progress only if the channel is marked with `ChanStatusLocalDataLoss` and the trigger is either `remoteCloseTrigger` or `breachCloseTrigger` (covering our bases)

Comment (Roasbeef) [reply contractcourt/channel_arbitrator.go:1064]: Yeah so my mental model is that eventually something will confirm, and we'll get a signal from the chain watcher that the channel has been force closed. Moving to the next state also means we won't try to broadcast again on start up either. We'll instead get the `StateContractClosed` signal. The state will be "unknown" so we'll pass in an empty commitment, like the SCB flow, then just play the commitment outputs. 

The addition of the `ErrForceCloseLocalDataLoss` error means we'll only ever do this if we had local data loss, so I think that's covered?

**Comment by Roasbeef** [comment]
> I think we should catch the error here to make sure we can finish startup,


True, we want to see have the channel arb active so it can get the on chain close signal.

**Comment by Roasbeef** [comment]
> And no advancing the state in channel arbitrator. 

Ah so you mean don't state step at all and instead wait for the on chain confirmation? Each time a block is mined though, we may end up triggering the go-to-chain decision, so I think we still need to catch this error and just have the state machine terminate as is. I updated the commit to just go back to `StateBroadcastCommit`, so it terminates there. Otherwise, in `StateDefault`, we'll do `checkLocalChainActions`, eventually deciding to force close again once the next block comes around.

**Comment by Roasbeef** [comment]
Pushed up a new commit:
  * test case added to ensure we start up 
  * modified to not advance to next state, just stay in `StateBroadcastCommit`

**Review by yyforyongyu — APPROVED** [review]
pending linter fix, otherwise LGTM🙏

I think going back to either `StateDefault` or `StateBroadcastCommit` works. Once in DLP mode, there isn't much we can do here but listening to remote force close and possible breach, and these two triggers can both be handled as long as we don't go beyong `StateBroadcastCommit`.

A future PR may wanna investigate and modify the `ChainArbitrator.Start` to never stop when one of the channel arbitrators fails to start,
https://github.com/lightningnetwork/lnd/blob/90effda090e752662500aa47eaed8b8619687e4e/contractcourt/chain_arbitrator.go#L707-L709

I think unless it's related to db failure, the system should start normally so it can operate on other channels.

**Review by Crypt-iQ — APPROVED** [review]

**Pull request merged**
